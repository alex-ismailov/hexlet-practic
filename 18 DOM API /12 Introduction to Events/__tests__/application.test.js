import fs from 'fs';
import path from 'path';
import { html } from 'js-beautify';

import run from '../src/application.js';

const fixuturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML);

beforeAll(() => {
  const initHtml = fs.readFileSync(path.join(fixuturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', () => {
  const element = document.querySelector('a[href="#profile"]');
  element.click();
  expect(getTree()).toMatchSnapshot();

  const element2 = document.querySelector('a[href="#settings"]');
  element2.click();
  expect(getTree()).toMatchSnapshot();

  const element3 = document.querySelector('a[href="#profile"]');
  element3.click();
  expect(getTree()).toMatchSnapshot();

  const element4 = document.querySelector('a[href="#profile2"]');
  element4.click();
  expect(getTree()).toMatchSnapshot();
});
