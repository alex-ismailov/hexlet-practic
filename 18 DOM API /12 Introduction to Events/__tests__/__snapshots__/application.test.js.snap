// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#home3\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile3\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane active\\" id=\\"home3\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile3\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

    <!-- Nav tabs -->
    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#home\\" role=\\"tab\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#profile\\" role=\\"tab\\">Profile</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#messages\\" role=\\"tab\\">Messages</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#settings\\" role=\\"tab\\">Settings</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane\\" id=\\"home\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane active\\" id=\\"profile\\" role=\\"tabpanel\\">
            Profile
        </div>
        <div class=\\"tab-pane\\" id=\\"messages\\" role=\\"tabpanel\\">
            Messages
        </div>
        <div class=\\"tab-pane\\" id=\\"settings\\" role=\\"tabpanel\\">
            Settings
        </div>
    </div>

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#home2\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile2\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane active\\" id=\\"home2\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile2\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

</div>"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#home3\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile3\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane active\\" id=\\"home3\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile3\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

    <!-- Nav tabs -->
    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#home\\" role=\\"tab\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile\\" role=\\"tab\\">Profile</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#messages\\" role=\\"tab\\">Messages</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#settings\\" role=\\"tab\\">Settings</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane\\" id=\\"home\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile\\" role=\\"tabpanel\\">
            Profile
        </div>
        <div class=\\"tab-pane\\" id=\\"messages\\" role=\\"tabpanel\\">
            Messages
        </div>
        <div class=\\"tab-pane active\\" id=\\"settings\\" role=\\"tabpanel\\">
            Settings
        </div>
    </div>

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#home2\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile2\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane active\\" id=\\"home2\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile2\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

</div>"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#home3\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile3\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane active\\" id=\\"home3\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile3\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

    <!-- Nav tabs -->
    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#home\\" role=\\"tab\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#profile\\" role=\\"tab\\">Profile</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#messages\\" role=\\"tab\\">Messages</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#settings\\" role=\\"tab\\">Settings</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane\\" id=\\"home\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane active\\" id=\\"profile\\" role=\\"tabpanel\\">
            Profile
        </div>
        <div class=\\"tab-pane\\" id=\\"messages\\" role=\\"tabpanel\\">
            Messages
        </div>
        <div class=\\"tab-pane\\" id=\\"settings\\" role=\\"tabpanel\\">
            Settings
        </div>
    </div>

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#home2\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile2\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane active\\" id=\\"home2\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile2\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

</div>"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#home3\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#profile3\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane active\\" id=\\"home3\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane\\" id=\\"profile3\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

    <!-- Nav tabs -->
    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#home\\" role=\\"tab\\">Home</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#profile\\" role=\\"tab\\">Profile</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#messages\\" role=\\"tab\\">Messages</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#settings\\" role=\\"tab\\">Settings</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane\\" id=\\"home\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane active\\" id=\\"profile\\" role=\\"tabpanel\\">
            Profile
        </div>
        <div class=\\"tab-pane\\" id=\\"messages\\" role=\\"tabpanel\\">
            Messages
        </div>
        <div class=\\"tab-pane\\" id=\\"settings\\" role=\\"tabpanel\\">
            Settings
        </div>
    </div>

    <ul class=\\"nav nav-tabs\\" role=\\"tablist\\">
        <li class=\\"nav-item\\">
            <a class=\\"nav-link\\" data-toggle=\\"tab\\" href=\\"#home2\\" role=\\"tab\\">Home 2</a>
        </li>
        <li class=\\"nav-item\\">
            <a class=\\"nav-link active\\" data-toggle=\\"tab\\" href=\\"#profile2\\" role=\\"tab\\">Profile 2</a>
        </li>
    </ul>

    <!-- Tab panes -->
    <div class=\\"tab-content m-3\\">
        <div class=\\"tab-pane\\" id=\\"home2\\" role=\\"tabpanel\\">
            Home
        </div>
        <div class=\\"tab-pane active\\" id=\\"profile2\\" role=\\"tabpanel\\">
            Profile
        </div>
    </div>

</div>"
`;
